cxx-qt-build/src/lib.rs

```diff
@@ -108,6 +108,16 @@ fn manifest_dir() -> String {
     manifest_dir
 }
 
+fn out_dir() -> String {
+    let mut out_dir = env::var("OUT_DIR").expect("Could not get out dir");
+    if cfg!(windows) {
+        out_dir = out_dir.replace('\\', "/");
+    }
+    out_dir += "/../../../..";
```

Comment:
Ah now I see why things later look strange, so we are abusing the OUT_DIR here, wonder if this has other implications and if we should go straight to the CLI tool for CMake to use :thinking: ( #/106 )

Reply:
Although this path is strange, the only way I know to place Rust's target-dir in the Qt build-dir without other CLI is calculate the target-dir through OUT_DIR. If there is a better way, please tell me.

Reply:
I worry that this will have problems when we reach cross compiling, eg `target/debug/build/<crate-hash>/out/` vs `target/aarch64-unknown-linux-gnu/debug/build/<crate-hash>/out/`.

What if you compared the OUT_DIR to the CARGO_MANIFEST_DIR could you then work out how many levels "up" it is from the OUT_DIR to the CARGO_MANIFEST_DIR..

Ah I think i see what's happening and why you wouldn't want to do that ^^, this is trying to put `cxx-qt-gen` into either `target/cxx-qt-gen` or `target/aarch64-unknown-linux-gnu/cxx-qt-gen` or where ever is specified ?

So then this probably a good first step until we build the CLI tool (which i'll write up the discussion for shortly).

Reply:
So this is probably fine for now, eventually we'll move to a CLI tool as being discussed here https://github.com/KDAB/cxx-qt/discussions/146